### DISCUSSION TITLE
Classes defined in SWC shown as error in text editor

### DISCUSSION THREAD

**Discussion by vync79** [BowlerHatLLC/vscode-as3mxml#626] [General]
Hello,
despite properly setting the swc libraries in the compilation path, it seems the vs code text editor cannot use the classes defined in the swc for runtime code completion and also shows undefined classes errors. 
Is there any way to configure the vscode extension to use swc libraries classes within the live editor without error and proper code completion ? 
Regards,
Vincent.

**Comment by joshtynjala** [comment]
You need to reference your .swc libraries in _asconfig.json_ using the [`library-path`](https://github.com/BowlerHatLLC/vscode-as3mxml/wiki/asconfig.json#library-path) compiler option.

**Reply by vync79** [reply]
Hi Josh,
this is what I tried on the first place.
Here is the content of the asconfig file I use.
As you can see the swc file is realted to a library you know quite well :-)
For any reason, all the Feathers classes appear as unknown in the vscode file editor whereas the project compiles and runs properly...

```
{
	"config": "airmobile",
	"compilerOptions": {
		"source-path": [
			"src"
		],
		"output": "xxx/xxx.swf",
		"library-path": [
			"lib/feathers.swc",
			"lib/AeonDesktopTheme.swc"
		]
	},
	"airOptions": {
		"files": [
			{
			  "file":"assets/icons",
			  "path": "icons"
			},
			{
			  "file": "assets/media",
			  "path": "media"
			}
		]
	},	
	"application": "src/xxx-app.xml",
	"mainClass": "MainXXX",
	"additionalOptions": "-compiler.keep-as3-metadata+=Enum,EditorHint"
}
```

**Reply by vync79** [reply]
Actually I found the root cause: in my project I have multiple asconfig files (asconfig-xxx.json, asconfig-xxx-64bit.json and asconfig.json) as I am generating multiple applications from the same project.
If I copy the Feathers swc in the library-path of the asconfig.json instead of the asconfig-xxx.json, it solves the issue but ultimately I do not want to include feathers in the application built using this asconfig.json file but only on the other ones...

It would be convenient to be able to define which one is the active asconfig file when working on the project but I guess it is quite possibly too specific to my setup...

Do you think reylying on which task is configured as the current defaut build task in vscode could be used to know which asconfig.json file is supposed to be used for the live code checking and completion in the editor ?

For instance, here is one of the build task I use which mention the specific asconfig file:
       {
            "type": "actionscript",
            "debug": true,
            "asconfig": "asconfig-xxx.json",
            "group": {
                "kind": "build",
                "isDefault": true
            },
            "problemMatcher": [],
            "label": "ActionScript: compile debug - asconfig-xxx.json"
        }
